File: cpp/1076.cpp
#include <iostream>
#include <unordered_set>

using namespace std;

int v;
bool graph[50][50];
unordered_set<int> visited;

void clear()
{
    for (int i = 0; i < 50; i++) {
        for (int j = 0; j < 50; j++) {
            graph[i][j] = false;
        }
    }
    visited.clear();
}

void add_edge(int a, int b)
{
    graph[a][b] = true;
    graph[b][a] = true;
}

int dfs(int start) 
{
    int edges = 0;
    visited.insert(start);
    for (int i = 0; i < v; i++) {
        if (graph[start][i] && visited.find(i) == visited.end()) {
            edges += dfs(i) + 1;
        }
    }
    return edges;
}

int main()
{
    int n, a, fr, to, start;
    cin >> n;
    for (n; n > 0; n--) {
        cin >> start >> v >> a;
        
        for (int i = 0; i < a; i++) {
            cin >> fr >> to;
            add_edge(fr, to);
        }
        cout << 2 * dfs(start) << '\n';
        clear();
    }
    return 0;
}

File: py/1076.py
class Graph:
    def __init__(self, v) -> None:
        self.graph = [[False for i in range(v)] for j in range(v)]
        self.v = v
        self.visited = set()

    def add_edge(self, a, b) -> None:
        self.graph[a][b] = True
        self.graph[b][a] = True

    def dfs(self, start) -> int:
        edges = 0
        v = self.v
        self.visited.add(start)
        for i in range(v):
            if self.graph[start][i] and i not in self.visited:
                edges += self.dfs(i) + 1
        return edges


def main() -> None:
    n = int(input())
    for _ in range(n):
        start = int(input())
        v, a = input().split()
        v, a = int(v), int(a)
        gr = Graph(v)
        for __ in range(a):
            fr, to = input().split()
            fr, to = int(fr), int(to)
            gr.add_edge(fr, to)
        print(2 * gr.dfs(start))


if __name__ == '__main__':
    main()
